app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User
+  include ActiveModel::Model
+
+  def self.find(username)
+    client = TradeTariffIdentity.cognito_client
+    response = client.admin_get_user({
+      user_pool_id: TradeTariffIdentity.cognito_user_pool_id,
+      username:,
+    })
+
+    User.new(username:, email: response.user_attributes.find { |attr| attr.name == "email" }&.value)
+  rescue Aws::CognitoIdentityProvider::Errors::UserNotFoundException
+    nil
```

Comment:
maybe some logging if we're not ready to do much more exception handling?

Reply:
Any other error will throw an exception so that will get logged.
Did you have something specific in mind?

Reply:
Hmm, I don't know. If we don't find them, cool but shouldn't we be able to feed that back? Unless this is an extra defense and we do some other validation elsewhere?

Reply:
If we don't find them then the API response will 404 "user not found", so the API caller will know.

Reply:
Ok cool. So that's fine!